core/computed/metrics/tbt-utils.js

```diff
@@ -6,6 +6,42 @@
 
 const BLOCKING_TIME_THRESHOLD = 50;
 
+/**
+ * @param {{start: number, end: number, duration: number}} event
+ * @param {number} startTimeMs
+ * @param {number} endTimeMs
+ * @param {{start: number, end: number, duration: number}} [topLevelEvent] Leave unset if `event` is top level. Has no effect if `event` has the same duration as `topLevelEvent`.
+ * @return {number}
+ */
+function calculateTbtImpactForEvent(event, startTimeMs, endTimeMs, topLevelEvent) {
+  let threshold = BLOCKING_TIME_THRESHOLD;
+  if (topLevelEvent) threshold *= (event.duration / topLevelEvent.duration);
+
+  // Early exit for small tasks, which should far outnumber long tasks.
+  if (event.duration < threshold) return 0;
+
+  // We only want to consider tasks that fall in our time range (FCP and TTI for navigations).
+  // FCP is picked as the lower bound because there is little risk of user input happening
+  // before FCP so Long Queuing Qelay regions do not harm user experience. Developers should be
+  // optimizing to reach FCP as fast as possible without having to worry about task lengths.
+  if (event.end < startTimeMs) return 0;
+
+  // TTI is picked as the upper bound because we want a well defined end point for page load.
+  if (event.start > endTimeMs) return 0;
```

Comment:
nit: These comments are better on the callsite IMO (or the function description for examples of how to use start/end time?) - In the body here they are out of place since the function itself is not in charge of finding these time points associated with the metrics mentioned in the comments.

(I know this was pre-existing code)